lib/commands/cloud-projects.ts

```diff
@@ -54,18 +59,37 @@ export class CloudExportProjectsCommand implements ICommand {
 		private $project: Project.IProject,
 		private $projectConstants: Project.IProjectConstants,
 		private $remoteProjectService: IRemoteProjectService,
-		private $server: Server.IServer) { }
+		private $server: Server.IServer,
+		private $prompter: IPrompter,
+		private $userDataStore: IUserDataStore) { }
 
-	allowedParameters: ICommandParameter[] = [new ProjectIdCommandParameter(this.$remoteProjectService)];
+	allowedParameters: ICommandParameter[] = [];
 
 	execute(args: string[]): IFuture<void> {
 		return (() => {
-			let name = this.$remoteProjectService.getProjectName(args[0]).wait();
-			this.doExportRemoteProject(name).wait();
+			let projectName: string;
+			let slnName = args[0] ? this.$remoteProjectService.getSolutionName(args[0]).wait() :
+									this.$prompter.promptForChoice("Select solution for which to list projects:", 
+												this.$remoteProjectService.getSolutions().wait().map(sln => sln.name)).wait();
+			
+			if(args[1]) {
+				projectName = this.$remoteProjectService.getProjectName(slnName, args[1]).wait();
+			} else {
+				let projectNames = this.$remoteProjectService.getProjectsForSolution(slnName).wait().map(sln => sln.Name);
+				if(projectNames.length === 1) {
+					projectName = projectNames[0];
+				} else if (projectNames.length > 1) {
+					projectName = this.$prompter.promptForChoice("Select project which you want to export:", projectNames).wait();
+				} else {
+					this.$errors.failWithoutHelp(`Solution ${slnName} does not have any projects.`);
```

Comment:
well you want to export a project, but there's no project - sounds like an error. I can change it to warn or logger.error and return from the method here.